extensions-contrib/compressed-bigdecimal/src/main/java/org/apache/druid/compressedbigdecimal/CompressedBigDecimal.java

```diff
@@ -266,15 +262,27 @@ protected static <S> int signumInternal(int size, S rhs, ToIntBiFunction<S, Inte
    * @see java.lang.Comparable#compareTo(java.lang.Object)
    */
   @Override
-  public int compareTo(CompressedBigDecimal<T> o)
+  public int compareTo(CompressedBigDecimal o)
   {
 
-    if (this.equals(o)) {
+    if (super.equals(o)) {
       return 0;
     }
     return this.toBigDecimal().compareTo(o.toBigDecimal());
   }
 
+  @Override
+  public int hashCode()
+  {
+    return super.hashCode();
```

Comment:
I'm a little suspect of these `super` calls in hashCode and equals as well.

Reply:
yea see above, i accidentally reverted a change to these